Scripts/Build-Phases/copy-secret.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+SOURCE_PATH=$1
+TARGET_PATH=$2
+
+USAGE="error: Usage $0 source_path target_path"
+
+if [[ -z ${SOURCE_PATH:+x} || -z ${TARGET_PATH:+x} ]]; then
+  echo $USAGE
+  exit 1
+fi
+
+git check-ignore "${TARGET_PATH}" > /dev/null
+GIT_IGNORE_EXIT_CODE=$?
+if [ $GIT_IGNORE_EXIT_CODE -ne 0 ]; then
+    echo "error: Attempting to store secret file in path that is not ignored by Git (${TARGET_PATH})."
+    exit 1
+fi
+
+if [ ! -f ${SOURCE_PATH} ]; then
+    echo "error: Unable to copy credentials. Could not find ${SOURCE_PATH}."
+    exit 1
+else
+    echo "Copying credentials..."
+    mkdir -p ${TARGET_FOLDER}
+    cp ${SOURCE_PATH} ${TARGET_PATH}
+fi
```

Comment:
If we like this approach to copy over secret `.swift` files, we could move this logic into the release toolkit.

Something I don't like, though, is that in its current form it is not very helpful for external contributors. 🤔

Reply:
An idea I have for a next iteration on this is to provide another parameter, maybe optional, to the script to fallback to if the source can't be found. That could be the value of the example file and the script could echo a warning:

> Could not find secret at `$SOURCE_PATH`. If you are a maintainer of this project, please ensure you [... instructions for `configure` ...]. In case you are an external contributor instead, I copied `$FALLBACK_PATH` to `$TARGET_PATH` to allow the build to proceed.

Another option could be to not copy the example file ourselves, to ensure everyone is aware of this, and instead print the suggestion as part of the error.

> Unable to copy credentials. Could not find `${SOURCE_PATH}`. If you are a maintainer of this project, [... instructions ...]. If you are an external contributor, copy `$FALLBACK_PATH` to `$TARGET_PATH` instead to allow the build to continue.

Reply:
Something else for consideration – adding an `example` file to the `.configure` file entry, so that it copies that one if the real one can't be decrypted?

I like the idea of sharing this with the release toolkit, but that puts a dependency on Ruby in every project (which isn't as nice).

Reply:
Regarding the extra Ruby dependency: dang I didn't consider that... 🤔 It shouldn't be hard to make it part of the upcoming Rust [`configure`](https://github.com/Automattic/configure/) itself, though.

Reply:
And now that I think about it, this is meaningful only for Xcode projects, which already depend on Ruby anyways, so we might be able to get away with it for some time. 🤷‍♂️